### PR TITLE
feat(esl-toggleable): revision v2 for ESLToggleable a11y manger and outside actions handling

### PR DESCRIPTION
**Pull request by ala-n**

### CODE REVIEW

File: src/modules/esl-toggleable/core/esl-toggleable.ts [hunk 1]
```diff
@@ -117,13 +117,14 @@ export class ESLToggleable extends ESLBaseElement {
   @attr({parser: parseBoolean, serializer: toBooleanAttribute}) public closeOnOutsideAction: boolean;
 
   /**
-   * Focus behavior. Available values:
+   * Acessability behavior. Available values:
```
Comment (dshovchko) [inline src/modules/esl-toggleable/core/esl-toggleable.ts:120]:
```suggestion
   * Accessibility behavior. Available values:
```

**Review by dshovchko — APPROVED** [review]

File: src/modules/esl-toggleable/core/esl-toggleable.ts [hunk 2]
```diff
@@ -361,8 +345,10 @@ export class ESLToggleable extends ESLBaseElement {
     const target = e.target as HTMLElement;
     // target is inside current toggleable
     if (this.contains(target)) return false;
-    // target is inside last activator
-    if (this.activator && this.activator.contains(target)) return false;
+
+    // target is inside chain of togleables
```
Comment (NastaLeo) [inline src/modules/esl-toggleable/core/esl-toggleable.ts:349]:
```suggestion
    // target is inside chain of toggleables
```

File: src/modules/esl-toggleable/core/esl-toggleable-manager.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+/** Focus flow behaviors */
+export type ESLA11yType = 'none' | 'autofocus' | 'popup' | 'dialog' | 'modal';
+
+let instance: ESLToggleableManager;
+/** Focus manager for toggleable instances. Singleton. */
+export class ESLToggleableManager {
+  /** Active toggleable */
+  protected active = new Set<ESLToggleable>();
+  /** Focus scopes stack. Manger observes only top level scope. */
```
Comment (NastaLeo) [inline src/modules/esl-toggleable/core/esl-toggleable-manager.ts:18]:
```suggestion
  /** Focus scopes stack. Manager observes only top level scope. */
```

**Review by NastaLeo — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

File: src/modules/esl-toggleable/core/esl-toggleable.ts [hunk 4]
```diff
@@ -117,13 +117,14 @@ export class ESLToggleable extends ESLBaseElement {
   @attr({parser: parseBoolean, serializer: toBooleanAttribute}) public closeOnOutsideAction: boolean;
 
   /**
-   * Focus behavior. Available values:
+   * Accessability behavior. Available values:
```
Comment (abarmina) [inline src/modules/esl-toggleable/core/esl-toggleable.ts:120]: extra cosmetic:

```suggestion
   * Accessibility behavior. Available values:
```

File: src/modules/esl-toggleable/README.md [hunk 5]
```diff
@@ -34,12 +34,13 @@ Use `ESLToggleableDispatcher.init()` to initialize (and bind) `ESLToggleableDisp
  - `close-on-esc` - Close the Toggleable on ESC keyboard event
  - `close-on-outside-action` - Close the Toggleable on a click/tap outside
 
- - `focus-behavior` - Focus flow behavior. <i class="badge badge-sup badge-success">new</i>  
+ - `a11y` - Accesibility behavior pattern. <i class="badge badge-sup badge-success">new</i>
```
Comment (abarmina) [inline src/modules/esl-toggleable/README.md:37]: cosmetic:

```suggestion
 - `a11y` - Accessibility behavior pattern. <i class="badge badge-sup badge-success">new</i>  
```

**Review by abarmina — APPROVED** [review]

**Pull request merged**